kolibri/plugins/coach/assets/src/vue/groups-page/group-section.vue

```diff
@@ -0,0 +1,199 @@
+      color="primary"
+      :has-dropdown="true"
+      ref="dropdownButton"
+      size="small">
+      <ui-menu slot="dropdown"
+        :options="menuOptions"
+        @select="handleSelection"
+        @close="close" />
+    </ui-button>
+
+    <table v-if="group.users.length">
+      <thead>
+        <tr>
+          <th class="col-checkbox">
+            <input type="checkbox"
+              :checked="allUsersAreSelected"
+              @change="toggleSelectAll">
+          </th>
+          <th class="col-name">{{ $tr('name') }}</th>
+          <th class="col-username">{{ $tr('username') }}</th>
+        </tr>
+      </thead>
+      <tbody>
+        <tr v-for="user in group.users"
+          :class="isSelected(user.id) ? 'selectedrow' : ''"
+          @click="toggleSelection(user.id)">
+          <td class="col-checkbox">
+            <input type="checkbox"
+              :id="user.id"
+              :value="user.id"
+              v-model="selectedUsers">
+          </td>
+          <td class="col-name"><strong>{{ user.full_name }}</strong></td>
+          <td class="col-username">{{ user.username }}</td>
+        </tr>
+      </tbody>
+    </table>
+    <p v-else>No Learners in this Group</p>
+  </div>
+
+</template>
+
+
+<script>
+
+  const groupActions = require('../../group-actions');
+
+  module.exports = {
+    $trNameSpace: 'coachGroupsTable',
+    $trs: {
+      numLearners: '{count, number, integer} {count, plural, one {Learner} other {Learners}}',
+      moveLearners: 'Move Learners',
+      renameGroup: 'Rename Group',
+      deleteGroup: 'Delete Group',
+      name: 'Name',
+      username: 'Username',
+      selected: 'Selected',
+    },
+    components: {
+      'icon-button': require('kolibri.coreVue.components.iconButton'),
+      'ui-button': require('keen-ui/src/UiButton'),
+      'ui-menu': require('keen-ui/src/UiMenu'),
+    },
+    props: {
+      group: {
+        type: Object,
+        required: true,
+      },
+      className: {
+        type: String,
+        required: true,
+      },
+      classId: {
+        type: String,
+        required: true,
+      },
+      isUngrouped: {
+        type: Boolean,
+        default: false,
+      },
+    },
+    data() {
+      return {
+        selectedUsers: [],
+      };
+    },
+    computed: {
+      menuOptions() {
+        return [{ label: this.$tr('renameGroup') }, { label: this.$tr('deleteGroup') }];
+      },
+      allUsersAreSelected() {
+        return (this.group.users.length === this.selectedUsers.length)
+          && (this.selectedUsers.length !== 0);
+      },
+    },
+    methods: {
+      handleSelection(selectedOption) {
+        switch (selectedOption.label) {
+          case (this.$tr('renameGroup')):
+            this.$emit('rename', this.group.name, this.group.id);
+            break;
+          case (this.$tr('deleteGroup')):
+            this.$emit('delete', this.group.name, this.group.id);
+            break;
+          default:
```

Comment:
Given there's no default case, I dunno if a switch is appropriate here.

I like switch statements.